### ISSUE TITLE
Convergence issue on min distance between line segment and triangle.

### ISSUE THREAD

**Issue by nhz2** [oxfordcontrol/Clarabel.jl#103]
I was using Clarabel and JuMP to test my code that analytically calculates the minimum distance between line segments and triangles in 3D. In random testing I found a line segment and triangle that doesn't converge with Clarabel but does converge with Ipopt. 

In this example the line segment and triangle are almost, but not quite parallel.

```julia
using JuMP
using LinearAlgebra
using Clarabel
using Ipopt

a = [
        [0.796, 0.768, 0.334], 
        [0.879, 0.227, 0.297],
    ]
b = [
        [0.322, 0.079, 0.305], 
        [0.486, 0.145, 0.792], 
        [0.194, 0.999, 0.333],
    ]

function refDist2(optimizer, points1,points2)
    model = Model(optimizer)
    set_optimizer_attribute(model, "max_iter", 2000)
    n1 = length(points1)
    n2 = length(points2)
    dims = length(points1[1])
    @assert length(points1[1])==length(points2[1])
    @variable(model, s[i = 1:n1] ≥ 0)
    @variable(model, t[i = 1:n2] ≥ 0)
    @constraint(model, sum(s[i] for i in 1:n1) == 1)
    @constraint(model, sum(t[i] for i in 1:n2) == 1)
    @expression(model, pt1[i = 1:dims], sum(s[j]*points1[j][i] for j in 1:n1))
    @expression(model, pt2[i = 1:dims], sum(t[j]*points2[j][i] for j in 1:n2))
    @expression(model, d[i = 1:dims], pt1[i]-pt2[i])
    @objective(model, Min, sum(d[i]*d[i] for i in 1:dims))
    optimize!(model)
    objective_value(model)
end

d2min_clarabel = refDist2(Clarabel.Optimizer, a, b)

d2min_ipopt = refDist2(Ipopt.Optimizer, a, b)

@show d2min_clarabel

@show d2min_ipopt
```
Clarabel finishes with:
```
1998   2.8110e-01   2.7930e-01  1.79e-03  8.19e-10  3.35e-09  6.50e-05  6.83e-63  4.80e-01  
1999   2.8015e-01   2.7941e-01  7.46e-04  2.88e-10  7.15e-09  3.75e-05  2.87e-63  6.22e-01  
2000   2.8114e-01   2.7931e-01  1.83e-03  8.09e-10  4.35e-09  7.41e-05  6.02e-63  5.26e-01  
---------------------------------------------------------------------------------------------
Terminated with status = iteration limit
solve time =  333ms
```

Ipopt finishes with:
```
   6  2.7969534e-01 0.00e+00 1.84e-11  -5.7 2.62e-03    -  1.00e+00 1.00e+00f  1
   7  2.7968859e-01 1.11e-16 9.22e-07  -8.6 4.20e-04    -  1.00e+00 9.92e-01f  1
   8  2.7968851e-01 0.00e+00 2.54e-14  -8.6 7.70e-06    -  1.00e+00 1.00e+00f  1

Number of Iterations....: 8

...

EXIT: Optimal Solution Found.
```

This is in Julia 1.8.1 with the following packages installed
```
(clarabelissue) pkg> st
Status `~/Desktop/clarabelissue/Project.toml`
  [61c947e1] Clarabel v0.3.0
  [b6b21f68] Ipopt v1.1.0
  [4076af6c] JuMP v1.3.0
```

**Comment by goulart-paul** [comment]
That is .... really odd.   I can't say I understand what is going on with this problem, but I note that disabling equilibration in the solver allows it to converge in 8 iterations.   If I add  this:
```julia
set_optimizer_attribute(model, "equilibrate_enable", false)
```

then I get this:
```
-------------------------------------------------------------
           Clarabel.jl v0.3.0  -  Clever Acronym              
                   (c) Paul Goulart                          
                University of Oxford, 2022                   
-------------------------------------------------------------

problem:
  variables     = 5
  constraints   = 7
  nnz(P)        = 15
  nnz(A)        = 10
  cones (total) = 2
    : Clarabel.Zero = 1,  numel = 2
    : Clarabel.Nonnegative = 1,  numel = 5

settings:
  linear algebra: direct / qdldl, precision: Float64
  max iter = 200, time limit = Inf,  max step = 0.990
  tol_feas = 1.0e-08, tol_abs = 1.0e-08, tol_rel = 1.0e-08,
  static reg : on, ϵ1 = 1.0e-07, ϵ2 = 4.9e-32
  dynamic reg: on, ϵ = 1.0e-13, δ = 2.0e-07
  iter refine: on, reltol = 1.0e-12, abstol = 1.0e-12, 
               max iter = 10, stop ratio = 5.0
  equilibrate: off, min_scale = 1.0e-04, max_scale = 1.0e+04
               max iter = 10

iter    pcost        dcost       gap       pres      dres      k/t        μ       step      
---------------------------------------------------------------------------------------------
  0   2.8732e-01  -2.8732e-01  5.75e-01  3.26e-01  6.92e-01  1.00e+00  1.00e+00   ------   
  1   2.8730e-01   2.8445e-01  2.85e-03  3.77e-02  9.85e-02  1.26e-01  1.52e-01  9.01e-01  
  2   2.8583e-01   2.8552e-01  3.01e-04  3.15e-03  8.12e-03  1.02e-02  1.32e-02  9.13e-01  
  3   2.8008e-01   2.7959e-01  4.85e-04  3.58e-04  9.55e-04  1.08e-03  1.88e-03  9.20e-01  
  4   2.8017e-01   2.7960e-01  5.70e-04  7.10e-05  1.91e-04  1.20e-04  5.94e-04  9.90e-01  
  5   2.7974e-01   2.7968e-01  5.97e-05  5.98e-06  1.61e-05  1.10e-05  5.76e-05  9.17e-01  
  6   2.7969e-01   2.7969e-01  4.63e-06  7.26e-08  1.96e-07  6.38e-07  3.14e-06  9.90e-01  
  7   2.7969e-01   2.7969e-01  6.48e-08  7.29e-10  1.96e-09  8.80e-09  4.30e-08  9.90e-01  
  8   2.7969e-01   2.7969e-01  6.51e-10  7.30e-12  1.96e-11  8.85e-11  4.32e-10  9.90e-01  
---------------------------------------------------------------------------------------------
Terminated with status = solved
solve time = 3.72ms
```

An end user shouldn't be expected to know to do that though, and in almost all cases the equilibration improves things.   I suggest to leave this open if you don't mind so that we can take a look at it in detail.

**Comment by goulart-paul** [comment]
Fixed in #114.   Thank you for reporting this error, since it exposed a fairly unusual edge case.
